### PR TITLE
DrawNode V2 

### PR DESCRIPTION
**Pull request by aismann**
## Describe your changes

Add all review parts. 
Please review again. Thanks.

### For each PR
- [ ] Add Copyright if it missed:   
      - `"Copyright (c) 2019-present Axmol Engine contributors (see AUTHORS.md)."`
- [x] I have performed a self-review of my code.
       
   Optional:
   - [ ] I have checked readme and add important infos to this PR.
   - [ ] I have added/adapted some tests too.
          
### For core/new feature PR
- [x] I have checked readme and add important infos to this PR.
- [x] I have added thorough tests.

### CODE REVIEW

**Comment by aismann** [comment]
@smilediver, @halx99 
Please review again. Thanks.

**Comment by halx99** [comment]
the 2nd test of `Node:Draw` will crash on android devices, and follow test has flick problem on android devices:
Screenshot_2024-09-01-22-33-11-177_org axmol cpp_tests

**Comment by halx99** [comment]
just download cpp-test apk from this PR latest build artifacts will reproduce above problems:

https://github.com/axmolengine/axmol/actions/runs/10653604025/artifacts/1879283194

**Comment by aismann** [comment]
> just download cpp-test apk from this PR latest build artifacts will reproduce above problems:
> 
> https://github.com/axmolengine/axmol/actions/runs/10653604025/artifacts/1879283194

I think its the tester itself which be not stable.
I will look.

**Comment by halx99** [comment]
> the 2nd test of `Node:Draw` will crash on android devices, and follow test has flick problem on android devices: Screenshot_2024-09-01-22-33-11-177_org axmol cpp_tests

The original draw node works well

**Comment by aismann** [comment]
@halx99 
Can you check the android issue again?
I will check it at home later

File: tests/cpp-tests/Source/DrawNodeTest/DrawNodeTest.cpp [hunk 1]
```diff
@@ -0,0 +1,2396 @@
...
+
+    drawNode->setScale(0.5);
+    drawNode->setPosition(center);
+
+    initSliders();
+    slider[sliderType::Thickness]->setEnabled(true);
+    slider[sliderType::Rotation]->setEnabled(true);
+
+
+    labelRound = Label::createWithTTF("DrawNode::Round", "fonts/Arial.ttf", 12);
```
Comment (halx99) [inline tests/cpp-tests/Source/DrawNodeTest/DrawNodeTest.cpp]:
```suggestion
    labelRound = Label::createWithTTF("DrawNode::Round", "fonts/arial.ttf", 12);
```

File: tests/cpp-tests/Source/DrawNodeTest/DrawNodeTest.cpp [hunk 2]
```diff
@@ -0,0 +1,2396 @@
...
+    addChild(labelRound, 1);
+    labelRound->setVisible(false);
+    labelSquare = Label::createWithTTF("DrawNode::Square", "fonts/Arial.ttf", 12);
```
Comment (halx99) [inline tests/cpp-tests/Source/DrawNodeTest/DrawNodeTest.cpp]:
```suggestion
    labelSquare = Label::createWithTTF("DrawNode::Square", "fonts/arial.ttf", 12);
```

File: tests/cpp-tests/Source/DrawNodeTest/DrawNodeTest.cpp [hunk 3]
```diff
@@ -0,0 +1,2396 @@
...
+    addChild(labelSquare, 1);
+    labelSquare->setVisible(false);
+    labelButt = Label::createWithTTF("DrawNode::Butt", "fonts/Arial.ttf", 12);
```
Comment (halx99) [inline tests/cpp-tests/Source/DrawNodeTest/DrawNodeTest.cpp]:
```suggestion
    labelButt = Label::createWithTTF("DrawNode::Butt", "fonts/arial.ttf", 12);
```

**Review by halx99 — CHANGES_REQUESTED** [review]

**Comment by halx99** [comment]
Could you also preserve old DrawNodeTests?

**Comment by aismann** [comment]
> Could you also preserve old DrawNodeTests?

Sure.
But the most be no longer usefull

File: tests/cpp-tests/Source/DrawNodeTest/DrawNodeTest.cpp (hunk 2 above)
Comment (aismann) [reply tests/cpp-tests/Source/DrawNodeTest/DrawNodeTest.cpp]: done

File: tests/cpp-tests/Source/DrawNodeTest/DrawNodeTest.cpp (hunk 1 above)
Comment (aismann) [reply tests/cpp-tests/Source/DrawNodeTest/DrawNodeTest.cpp]: done

**Comment by halx99** [comment]
> > Could you also preserve old DrawNodeTests?
> 
> Sure. But the most be no longer usefull

I just want to see all old tests works well

**Comment by halx99** [comment]
This test seems dead loop on android device:
Screenshot_2024-09-04-00-46-21-107_org axmol cpp_tests

**Comment by aismann** [comment]
@halx99 
Can you test it again?
Here on my device its works well (used: build / android (armv7))

**Comment by halx99** [comment]
> This test seems dead loop on android device: Screenshot_2024-09-04-00-46-21-107_org axmol cpp_tests

issue still exist, when I goto the test, I can't get any response when I click any button

**Comment by aismann** [comment]
> > This test seems dead loop on android device: Screenshot_2024-09-04-00-46-21-107_org axmol cpp_tests
> 
> issue still exist, when I goto the test, I can't get any response when I click any button

armv7 also?

**Comment by halx99** [comment]
> > > This test seems dead loop on android device: Screenshot_2024-09-04-00-46-21-107_org axmol cpp_tests
> > 
> > 
> > issue still exist, when I goto the test, I can't get any response when I click any button
> 
> armv7 also?

Yes

**Comment by aismann** [comment]
Can you try to reduce the sizes?
static float WID = 400;
    static float HIG = 600


I will remove this test from android devices

WID is  also on renderline again

**Comment by aismann** [comment]
@halx99 
Please review again. its all done and ready for merge

**Review by halx99 — APPROVED** [review]

**Comment by halx99** [comment]
@aismann does the folder https://github.com/axmolengine/axmol/tree/dev/extensions/DrawNodeEx/ can be deleted?

**Pull request merged**